### PR TITLE
[Triton-MLIR] Add bfloat16 support for AMDGPU.

### PR DESCRIPTION
**Pull request by rsanthanam-amd**
Add code to convert bfloat16 to and from float32 to enable elementwise op support for bfloat16.

This was done using LLVM intrinsics in order to avoid using the GCNBuilder construct.

### CODE REVIEW

**Review by whchung — CHANGES_REQUESTED** [review]
Please provide the description. And perhaps update "test_core_amd.py" if there are previously failing tests which are now passing with the PR.

File: lib/Conversion/TritonGPUToLLVM/Utility.h [hunk 1]
```diff
@@ -13,13 +13,17 @@
 #define inttoptr(...) rewriter.create<LLVM::IntToPtrOp>(loc, __VA_ARGS__)
 #define ptrtoint(...) rewriter.create<LLVM::PtrToIntOp>(loc, __VA_ARGS__)
 #define zext(...) rewriter.create<LLVM::ZExtOp>(loc, __VA_ARGS__)
+#define trunc(...) rewriter.create<LLVM::TruncOp>(loc, __VA_ARGS__)
```
Comment (whchung) [inline lib/Conversion/TritonGPUToLLVM/Utility.h:16]: @rsanthanam-amd changes made in this file doesn't seem related with bfloat16 support to me. could you help elaborate a bit?

Comment (rsanthanam-amd) [reply lib/Conversion/TritonGPUToLLVM/Utility.h:16]: i added these shorthand macros to this file because they were useful for the AMDGPU bfloat16 implementation and also because others may find them useful in future contexts.

for example, the zext macro was already there and it was useful for this PR.

In a similar fashion, I added trunc() and the shift macros.

from ElementwiseOpToLLVM.cpp:

auto shifted = shl(i32_ty, as_int32, i32_val(16));

// ...

auto shifted = lshr(i32_ty, as_int32, i32_val(16));
auto truncated = trunc(i16_ty, shifted);

**Comment by rsanthanam-amd** [comment]
> Please provide the description. And perhaps update "test_core_amd.py" if there are previously failing tests which are now passing with the PR.

I enabled bfloat16 for test_bin_op in test_core_amd.py.

**Review by whchung — APPROVED** [review]
